lib/jekyll/commands/serve.rb

```diff
@@ -98,17 +98,22 @@ def webrick_options(config)
 
           opts
         end
+        
+        # Allows files to be routed in a pretty URL in both default format
+        # and in custom page/index.html format and while doing so takes into
+        # consideration importance of blog.html > blog/ but not > blog/index.html
+        # because you could have URL's like blog.html, blog/archive/page.html
+        # and in a normal circumstance blog/ would be greater than blog.html
+        # breaking your entire site when you are playing around, and I
+        # don't think you really want that to happen when testing do you?
 
-        # Custom WEBrick FileHandler servlet for serving "/file.html" at "/file"
-        # when no exact match is found. This mirrors the behavior of GitHub
-        # Pages and many static web server configs.
         def custom_file_handler
           Class.new WEBrick::HTTPServlet::FileHandler do
             def search_file(req, res, basename)
-              if file = super
-                file
+              if (file = super) || (file = super req, res, "#{basename}.html")
+                then file
               else
-                super(req, res, "#{basename}.html")
+                return ".html"
```

Comment:
That's actually a feature that takes advantage of what might be (a bug?) in Webrick.  If you  return ".html" it will actually concatinate the path with ".html" so if you visit /blog and it knows you are at /blog and you return ".html" it will give you "blog.html"